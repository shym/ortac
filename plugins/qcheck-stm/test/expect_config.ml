open Ortac_qcheck_stm

let underline str =
  print_endline str;
  print_endline (String.map (Fun.const '=') str)

let pp = Reserr.pp (Fmt.pair Fmt.nop Config.dump) Fmt.stdout
let () = underline "test with existing sut type:"
let () = Config.init "lib.mli" "good_init" "sut" |> pp
let () = print_newline ()
let () = underline "test with non-existing sut type:"
let () = Config.init "lib.mli" "good_init" "absent" |> pp
let () = print_newline ()
let () = underline "test with a wrong type for init function:"
let () = Config.init "lib.mli" "bad_init" "sut" |> pp
let () = print_newline ()
let () = underline "test with an absent init function:"
let () = Config.init "lib.mli" "absent" "sut" |> pp
let () = print_newline ()
let () = underline "test with not well formed type expression:"
let () = Config.init "lib.mli" "" "a-t" |> pp
let () = print_newline ()
let () = underline "test with non instantiated sut type parameter:"
let () = Config.init "lib.mli" "" "'a t" |> pp
let () = print_newline ()
let () = underline "test with arrow type as sut:"
let () = Config.init "lib.mli" "" "int -> int" |> pp
