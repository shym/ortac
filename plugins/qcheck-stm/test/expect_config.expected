test with existing sut type:
============================

init: good_init; sut: sut

test with non-existing sut type:
================================

File "_none_", line 0, characters -1--1:
Error: Type `absent' is not declared in the module.

test with a wrong type for init function:
=========================================

File "_none_", line 0, characters -1--1:
Error: Function `bad_init' is not declared in the module.

test with an absent init function:
==================================

File "_none_", line 0, characters -1--1:
Error: Function `absent' is not declared in the module.

test with not well formed type expression:
==========================================

File "_none_", line 0, characters -1--1:
Error: `a-t' is not a well formed type expression.

test with non instantiated sut type parameter:
==============================================

File "_none_", line 1, characters -1--1:
Error: Type parameter of the system under test should be instantiated.

test with arrow type as sut:
============================

File "_none_", line 1, characters -1--1:
Error: The type given for the system under test is not supported.

